### PR TITLE
Refactor Headers. Util to include Headers into outgoing context

### PR DESCRIPTION
**Pull request by addudko**

### CODE REVIEW

File: auth/jwt.go [hunk 1]
```diff
@@ -13,6 +13,9 @@ const (
 	// MultiTenancyField the field name for a specific tenant
 	MultiTenancyField = "account_id"
 
+	// AuthorizationHeader contains information about the header value for the token
+	AuthorizationHeader = "Authorization"
```
Comment (drewwells) [inline auth/jwt.go:17]: 👍🏻

File: gateway/header.go [hunk 2]
```diff
@@ -12,6 +12,33 @@ import (
...
+		"x-geo-country-code",
+		"x-geo-country-name",
+		"x-geo-region-code",
+		"x-geo-region-name",
+		"x-geo-city-name",
+		"x-geo-postal-code",
+		"x-geo-latitude",
+		"x-geo-longitude",
+	}
+}
```
Comment (drewwells) [inline gateway/header.go:30]: Why is this a func? If it's meant to make it extendable, a variable would just as well

```suggestion
var GetGeoHeaders = []string{
		"x-geo-org",
		"x-geo-country-code",
		"x-geo-country-name",
		"x-geo-region-code",
		"x-geo-region-name",
		"x-geo-city-name",
		"x-geo-postal-code",
		"x-geo-latitude",
		"x-geo-longitude",
}
```

File: gateway/header.go [hunk 3]
```diff
@@ -12,6 +12,33 @@ import (
...
+
+// GetXB3Headers returns a slice of x-b3- headers.
+func GetXB3Headers() []string {
+	return []string{
+		"x-b3-traceid",
+		"x-b3-parentspanid",
+		"x-b3-spanid",
+		"x-b3-sampled",
+	}
+}
```
Comment (drewwells) [inline gateway/header.go:40]: Same comment
```suggestion
var GetXB3Headers = []string{
		"x-b3-traceid",
		"x-b3-parentspanid",
		"x-b3-spanid",
		"x-b3-sampled",
	}
}
```

File: gateway/header.go [hunk 4]
```diff
@@ -116,17 +143,7 @@ func handleForwardResponseTrailer(w http.ResponseWriter, md runtime.ServerMetada
...
-		"x-geo-country-code",
-		"x-geo-country-name",
-		"x-geo-region-code",
-		"x-geo-region-name",
-		"x-geo-city-name",
-		"x-geo-postal-code",
-		"x-geo-latitude",
-		"x-geo-longitude",
-	)
+	return ExtendedDefaultHeaderMatcher(GetGeoHeaders()...)
```
Comment (drewwells) [inline gateway/header.go:146]:
```suggestion
	return ExtendedDefaultHeaderMatcher(GetGeoHeaders...)
```

File: gateway/header.go [hunk 5]
```diff
@@ -136,12 +153,7 @@ func RequestIDHeaderMatcher() runtime.HeaderMatcherFunc {
 
 // TracingHeaderMatcher tracing headers
 func TracingHeaderMatcher() runtime.HeaderMatcherFunc {
-	return ExtendedDefaultHeaderMatcher(
-		"x-b3-traceid",
-		"x-b3-parentspanid",
-		"x-b3-spanid",
-		"x-b3-sampled",
-	)
+	return ExtendedDefaultHeaderMatcher(GetXB3Headers()...)
```
Comment (drewwells) [inline gateway/header.go:156]:
```suggestion
	return ExtendedDefaultHeaderMatcher(GetXB3Headers...)
```

File: util/context.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	"context"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/infobloxopen/atlas-app-toolkit/auth"
+	"github.com/infobloxopen/atlas-app-toolkit/gateway"
+	"github.com/infobloxopen/atlas-app-toolkit/requestid"
+)
+
+// ForwardContext set to outgoing context request_id, auth_token and X-Forwarded-For header value
+func ForwardContext(ctx context.Context) context.Context {
```
Comment (drewwells) [inline util/context.go:13]: I think OutgoingContext would be a better name here. These are smaller helpers around metadata.NewOutgoingContext()

**Review by drewwells — APPROVED** [review]
Few comments, looks fine.

File: gateway/header.go (hunk 2 above)
Comment (addudko) [reply gateway/header.go:30]: The reason why I made it as a func is assuming that this headers are meant to be constant. Such approach ensures that there is no way we can get mutated value.

File: util/context.go (hunk 6 above)
Comment (addudko) [reply util/context.go]: Agree

**Review by Evgeniy-L — APPROVED** [review]

**Pull request merged**